bitmovin-analytics-datasource/src/datasource.ts

```diff
@@ -41,11 +45,32 @@ export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
     return { data };
   }
 
-  async testDatasource() {
-    // Implement a health check for your data source.
-    return {
-      status: 'success',
-      message: 'Success',
+  async request(url: string, method: string) {
+    const options = {
+      url: this.baseUrl + url,
+      headers: { 'X-Api-Key': this.apiKey },
+      method: method,
     };
+    const response = getBackendSrv().fetch(options);
+
+    return lastValueFrom(response);
+  }
+
+  async testDatasource() {
+    const response = await this.request('/analytics/licenses', 'GET');
+
+    if (response.status === 200) {
+      return {
+        status: 'success',
+        message: 'Datasource setup successfully.',
+      };
+    } else {
+      return {
+        status: 'error',
+        message: response.statusText
+          ? response.statusText
+          : 'An unexpected error occurred. Datasource not setup correctly.',
+      };
```

Comment:
Have you checked what's inside this `response.statusText` are there more fields describing the response?
If it is anyhow possible it would be awesome to have the correlationId in the message.
Just brings so much value when debugging

Reply:
Changed the errorMessages to include more details if available

![image](https://github.com/bitmovin/analytics-grafana-datasource/assets/48029745/527c92a5-6530-4a35-8346-c03f211f3a97)

![image](https://github.com/bitmovin/analytics-grafana-datasource/assets/48029745/12599573-53a0-4f72-91e3-4199c2460ad0)